calico/felix/frules.py

```diff
@@ -409,7 +409,9 @@ def interface_to_chain_suffix(config, iface_name):
     :param iface_name: The interface name
     :returns string: the suffix (shortened if necessary)
     """
-    suffix = iface_name.replace(config.IFACE_PREFIX, "", 1)
-    # The suffix is surely not very long, but make sure.
-    suffix = futils.uniquely_shorten(suffix, 16)
-    return suffix
+    for prefix in config.IFACE_PREFIX:
+        if iface_name.startswith(prefix):
+            iface_name = iface_name[len(prefix):]
+            break
```

Comment:
Ok, I got your idea, but according to your example this function will fail if any two prefixes have at least one common letter at the beginning. What if this function will search for the longest common prefix and remove it from the interface name.